### DISCUSSION TITLE
Lesson 4: Web3.py Simple Storage -  when i run "python deploy.py.

### DISCUSSION THREAD

**Discussion by Tundekuzco** [smartcontractkit/full-blockchain-solidity-course-py#1065] [Q&A] [answered]
when i run "python deploy.py in the terminal I get this message 

```Python
from solcx import compile_standard

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

# Compile Our Solidity

compiled_sol = compile_standard(
    {
        "language": "solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "ebm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

print(compiled_sol)
```
`
"PS C:\Users\tunde\demos\web3_py_simple_storage> python deploy.py
INFO: Could not find files for the given pattern(s).
Traceback (most recent call last):
    compiled_sol = compile_standard(
  File "C:\Python310\lib\site-packages\solcx\main.py", line 371, in compile_standard
    solc_binary = get_executable(solc_version)
  File "C:\Python310\lib\site-packages\solcx\install.py", line 194, in get_executable
    raise SolcNotInstalled(
solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed. Use solcx.install_solc('0.6.0') to install.

any help please

**Comment by n4n0b1t3** [comment]
Please check out the [chronological-issues-from-video.md for lesson 4](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4)
If this already solves your problem please mark this post as answered.
If this does not solve the issue, we'd like to see more information about your issue. Please refer to [how to improve your posts](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/980) before you write your question, this makes it easier for the community to help you.

**Comment by cromewar** [comment, marked as answer]
Hello @Tundekuzco 
Here:
`"language": "solidity",`

Solidity must be upper case:

`"language": "Solidity",`

Also:

1. you don't have installed py-solc-x you must install it trough pip: https://pypi.org/project/py-solc-x/
2. You are using python 3.10, please don't that version has problems with web3 libraries and specially with solc, use python 3.8 instead on a virtualenv
3. Don't forget to select correct python interpreter on vscode (bottom left corner).
